app/src/main/java/com/example/cardgame/GameplayFragment.kt

```diff
@@ -335,69 +217,105 @@ class GameplayFragment : Fragment() {
+                val strRes = Gson().fromJson<ResponseForNewCard>(
+                    result,
+                    ResponseForNewCard::class.java
+                )
+
+                cardInfo = strRes.cards[0]
+            } else {
+                Log.d(TAG, "Response unsuccessful")
             }
-        })
+        } catch (e: Exception) {
+            Log.e(TAG, "Draw card was unsuccessful due to network request fail")
+        }
+        cardInfo
     }
 
-    //toDo I'll probably delete this method
-    private fun callApiForNextCard(root: View): Int {
-        val client = OkHttpClient()
-        val deckApi = "https://deckofcardsapi.com/api/deck/$deckId/draw/?count=1"
-        val request = Request.Builder().url(deckApi).build()
+    private fun drawNewCard(shouldCompare: Boolean, op: (Int, Int) -> Boolean) {
+        disposable?.dispose()
+        disposable = getCardObservable()
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe { cardInfo ->
 
-        client.newCall(request).enqueue(object : Callback {
-            override fun onFailure(call: Call, e: IOException) {
-                e.printStackTrace()
-                Log.d(TAG, "REQUEST FOR NEW CARD FAILED!", e)
-            }
+                cardInfo?.apply {
+
+                    currentImage = image
+                    returnCard = Consts.indexOf(value)
 
-            override fun onResponse(call: Call, response: Response) {
-                if(response.isSuccessful) {
-                    val result = response.body()!!.string()
-                    val strRes = Gson().fromJson<ResponseForNewCard>(result,
-                        ResponseForNewCard::class.java)
-                    val image = strRes.cards[0].image
-                    val cardValue = strRes.cards[0].value
-                    returnCard = Consts.indexOf(cardValue)
                     activity?.runOnUiThread {
-                        val imgView = root.findViewById<ImageView>(R.id.ivCurrentCard)
+
+                        val imgBack = root!!.findViewById<ImageView>(R.id.ivCurrentCardBack)
+                        val imgView = root!!.findViewById<ImageView>(R.id.ivCurrentCard)
+                        val easyFlipView = root!!.findViewById<EasyFlipView>(R.id.easyFlipView)
+
+                        Picasso.get().load(R.drawable.back_of_a_card).into(imgBack)
+
+                        easyFlipView.flipDuration = 500
+                        easyFlipView.setFlipTypeFromLeft()
+                        easyFlipView.flipTheView(true)
+
+                        easyFlipView.flipDuration = 2000
+                        easyFlipView.setFlipTypeFromLeft()
+                        easyFlipView.flipTheView(true)
+
                         Picasso.get().load(image).into(imgView)
+
+                        val tv = root!!.findViewById<TextView>(R.id.tvScore)
+                        tv.text = currentScore.toString()
                     }
```

Comment:
Move the UI related code in separate method. You'll need to do it anyway once you create the view model. Also create a companion object for this class and add the hardcoded consts there.